ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/SponsoredInteraction.java

```diff
@@ -66,13 +66,17 @@ public static Builder builder() {
    * it will build one based on `source` and `interactionType` fields.
    * @return String
    */
-  public String getDocumentType() {
-    if (getOriginalDocType() != null) {
+  public String getDocumentType(boolean derived) {
```

Comment:
The boolean here makes it very hard to understand intent when this method gets called. I'd prefer a separate method called `getDerivedDocumentType`. And maybe get rid of `getDocumentType()` and instead rely on explicitly calling `getOriginalDocumentType`

Reply:
yeah that's better. good call